marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -114,4 +121,41 @@ private String convertRowToJSONString(InternalRow record) {
         return jsonObjectWriter.toString();
     }
 
+    protected JsonNode determineIngestionEndpointParams(Map<String, String> params) {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        ObjectNode endpointParams;
+        if (params.containsKey("ingestendpointparams")) {
+            try {
+                endpointParams = (ObjectNode)objectMapper.readTree(params.get("ingestendpointparams"));
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to parse ingestendpointparams, cause: " + e.getMessage(), e);
+            }
+        } else {
+            endpointParams = objectMapper.createObjectNode();
+        }
+
+        boolean doesNotHaveApiPath = (!endpointParams.hasNonNull("apiPath") || StringUtils.isEmpty(endpointParams.get("apiPath").asText()));
```

Comment:
endpointParams.get("apiPath").asText() will throw s NPE

Reply:
An NPE won't occur because if apiPath were null or did not exist, then the first part of the conditional would be true and then Java won't evaluate the second part.